src/client-agent/notify.c

```diff
@@ -128,74 +140,32 @@ void run_notify()
         }
     }
 
-    rand_keepalive_str2(keep_alive_random, KEEPALIVE_SIZE);
-
-#if defined (__linux__) || defined (__MACH__)
-    char *agent_ip;
-    int sock;
-    char label_ip[60];
-    int i;
-    os_calloc(IPSIZE + 1,sizeof(char),agent_ip);
-
-    for (i = SOCK_ATTEMPTS; i > 0; --i) {
-        if (sock = control_check_connection(), sock >= 0) {
-            if (OS_SendUnix(sock, agent_ip, IPSIZE) < 0) {
-                merror("Error sending msg to control socket (%d) %s", errno, strerror(errno));
-            }
-            else{
-                if(OS_RecvUnix(sock, IPSIZE, agent_ip) == 0){
-                    merror("Error receiving msg from control socket (%d) %s", errno, strerror(errno));
-                    *agent_ip = '\0';
-                }
-                else{
-                    snprintf(label_ip,sizeof label_ip,"#\"_agent_ip\":%s", agent_ip);
-                }
-            }
-
-            close(sock);
-            break;
-        } else {
-            mdebug2("Control module not yet available. Remaining attempts: %d", i - 1);
-            sleep(1);
-        }
-    }
-
-    if(sock < 0) {
-        merror("Unable to bind to socket '%s': (%d) %s.", CONTROL_SOCK, errno, strerror(errno));
-    }
+    agent_ip = get_agent_ip();
 
     /* Create message */
-    if(*agent_ip && strcmp(agent_ip,"Err")){
+    if(agent_ip && strcmp(agent_ip, "Err")) {
+        char label_ip[60];
+        snprintf(label_ip, sizeof label_ip, "#\"_agent_ip\":%s", agent_ip);
         if ((File_DateofChange(AGENTCONFIGINT) > 0 ) &&
                 (OS_MD5_File(AGENTCONFIGINT, md5sum, OS_TEXT) == 0)) {
-            snprintf(tmp_msg, OS_MAXSTR - OS_HEADER_SIZE, "#!-%s / %s\n%s%s%s\n%s",
-                    getuname(), md5sum, tmp_labels, shared_files, label_ip, keep_alive_random);
+            snprintf(tmp_msg, OS_MAXSTR - OS_HEADER_SIZE, "#!-%s / %s\n%s%s%s\n",
```

Comment:
`tmp_msg` string could be formed by truncated strings as its size can be smaller than the sum of the strings concatenated to form it. Please adjust the size of these strings.